sklearn/preprocessing/_encoders.py

```diff
@@ -731,6 +765,14 @@ def inverse_transform(self, X):
 
         for i in range(n_features):
             labels = X[:, i].astype('int64', copy=False)
-            X_tr[:, i] = self.categories_[i][labels]
+            # set unknown values to None
+            if self.handle_unknown == 'use_encoded_value':
+                unknown_labels = labels == self.unknown_value
+                X_tr[:, i] = self.categories_[i][np.where(
+                    unknown_labels, 0, labels)]
+                X_tr = X_tr.astype(object, copy=False)
```

Comment:
Is there a way to have only one call `astype(object)` similar to

https://github.com/scikit-learn/scikit-learn/blob/ae0d5355160f23f23db4121e6a1fa93c75111b86/sklearn/preprocessing/_encoders.py#L549-L556

In that cases the unknown masks are stored in a dictionary and the `None` are placed in `X_tr` after the fact.

Reply:
Yeah, that might be faster.